analytics_data_api/v0/views/course_summaries.py

```diff
@@ -37,101 +33,100 @@ class CourseSummariesView(generics.ListAPIView):
             * count_change_7_days: Total difference in enrollment counts over the past 7 days across modes.
             * enrollment_modes: For each enrollment mode, the count, cumulative_count, and count_change_7_days.
             * created: The date the counts were computed.
+            * programs: List of program IDs that this course is a part of.
 
     **Parameters**
 
         Results can be filed to the course IDs specified or limited to the fields.
 
         course_ids -- The comma-separated course identifiers for which summaries are requested.
             For example, 'edX/DemoX/Demo_Course,course-v1:edX+DemoX+Demo_2016'.  Default is to
-            return call courses.
+            return all courses.
         fields -- The comma-separated fields to return in the response.
-            For example, 'course_id,created_mode'.  Default is to return all fields.
-
-        fields -- Fields to include in response.  Default is all.
+            For example, 'course_id,created'.  Default is to return all fields.
+        exclude -- The comma-separated fields to exclude in the response.
+            For example, 'course_id,created'.  Default is to exclude the programs array.
+        programs -- If included in the query parameters, will find each courses' program IDs
+            and include them in the response.
     """
-
-    course_ids = None
-    fields = None
     serializer_class = serializers.CourseMetaSummaryEnrollmentSerializer
+    programs_serializer_class = serializers.CourseProgramMetadataSerializer
     model = models.CourseMetaSummaryEnrollment
-
-    def get_serializer(self, *args, **kwargs):
-        kwargs.update({
-            'context': self.get_serializer_context(),
-            'fields': self.fields,
-        })
-        return self.get_serializer_class()(*args, **kwargs)
+    model_id = 'course_id'
+    programs_model = models.CourseProgramMetadata
+    count_fields = ('count', 'cumulative_count', 'count_change_7_days')  # are initialized to 0 by default
+    summary_meta_fields = ['catalog_course_title', 'catalog_course', 'start_time', 'end_time',
+                           'pacing_type', 'availability']  # fields to extract from summary model
 
     def get(self, request, *args, **kwargs):
         query_params = self.request.query_params
-        self.fields = split_query_argument(query_params.get('fields'))
-        self.course_ids = split_query_argument(query_params.get('course_ids'))
-        if self.course_ids is not None:
-            for course_id in self.course_ids:
-                validate_course_id(course_id)
-
-        return super(CourseSummariesView, self).get(request, *args, **kwargs)
-
-    def default_summary(self, course_id, count_fields):
+        programs = split_query_argument(query_params.get('programs'))
+        if not programs:
+            self.always_exclude = ['programs']
```

Comment:
It might be a little safer to append "programs" in case `always_exclude` was set.